src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    public static final String TITLE = "title";
+    public static final String PRICE = "price";
+    public static final String ID = "id";
+    private static final int MIN_AFFECTED_ROWS = 1;
```

Comment:
This value represents 2 meanings, first is the minimum number of affected rows, and second one is title number in create method line 31. And it is a bit confusing to write: ' statement.setString(MIN_AFFECTED_ROWS, book.getTitle());' I suggest creating separate constant with 1 value for title number or just leave title number as plain int. I prefere second way